fix: serve always only the latest objects

harshavardhana opened the pull request:
## Description
fix: serve always only the latest objects

## Motivation and Context
due to a historic bug, it is possible that
some objects might exist on multiple pools,
rely on ModTime to return the correct pool.

## How to test this PR?
Create two distributed setups and then 
combine them by creating duplicate objects

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Optimization (provides speedup with no functional changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [ ] Documentation updated
- [ ] Unit tests added/updated

vadmeste commented on cmd/erasure-server-pool.go:
```diff
@@ -272,10 +283,37 @@ func (z *erasureServerPools) getPoolIdxExisting(ctx context.Context, bucket, obj
+			if isErrMethodNotAllowed(errs[i]) || isErrVersionNotFound(errs[i]) {
+				// No object exists or its a delete marker,
+				// check objInfo to confirm.
+				if objInfos[i].DeleteMarker && objInfos[i].Name != "" {
```
Can objInfos[i] be valid when errs[i] is not nil ?

harshavardhana replied:
yes @vadmeste for delete markers

harshavardhana committed:
fix: serve always only the latest objects

due to a historic bug, it is possible that
some objects might exist on multiple pools,
rely on ModTime to return the correct pool.

kannappanr reviewed (approved):
LGTM

poornas reviewed (approved):
LGTM
